modules/cloud-dataplex/main.tf

```diff
@@ -52,28 +82,42 @@ resource "google_dataplex_zone" "basic_zone" {
   resource_spec {
     location_type = var.location_type
   }
+}
 
-  project = var.project_id
+resource "google_dataplex_zone_iam_binding" "binding" {
+  for_each = {
+    for zone_role in local.zone_iam : "${zone_role.zone}-${zone_role.role}" => zone_role
+  }
+  project       = var.project_id
+  location      = var.region
+  lake          = local.zone_resources[each.value.zone].lake
+  dataplex_zone = local.zone_resources[each.value.zone].name
```

Comment:
Doesn't this equals to `each.value.zone`? (and then `zone_resources` are not needed anymore?)

Reply:
I created `zone_resources` variable to guarantee dependency between `google_dataplex_zone` and `google_dataplex_zone_iam_binding` and avoid `depends_on`. Do you see any better way?

Reply:
Yes, you're right. Though I guess, I'd still prefer direct `google_dataplex_zone.zone[each.value.zone].name` here. And as zone is dependent on lake, then no direct dependency on lake is required.